packages/@aws-cdk/cdk/lib/cloudformation/stack.ts

```diff
@@ -440,25 +444,15 @@ export class Stack extends Construct {
     }
     return false;
   }
-}
 
-function merge(template: any, part: any) {
-  for (const section of Object.keys(part)) {
-    const src = part[section];
-
-    // create top-level section if it doesn't exist
-    let dest = template[section];
-    if (!dest) {
-      template[section] = dest = src;
-    } else {
-      // add all entities from source section to destination section
-      for (const id of Object.keys(src)) {
-        if (id in dest) {
-          throw new Error(`section '${section}' already contains '${id}'`);
-        }
-        dest[id] = src[id];
-      }
-    }
+  /**
+   * Calculcate the stack name based on the construct path
+   */
+  private calculateStackName() {
+    return this.node.rootPath()
+        .filter(c => !App.isDefaultApp(c))
+        .map(c => c.node.id)
+        .join('-');
```

Comment:
Am I reading this wrong? `rootPath` removes the first element so it won't include the App? Your tests pass so I assume so :)

```typescript
public rootPath(): IConstruct[] {
  const ancestors = this.ancestors();
  ancestors.shift();
  return ancestors;
}
```

Reply:
`Program` is the actual root of the construct tree, so that's the one that gets cut.